### PR TITLE
Add the option to configure the base-url of the call used by Jibri to…

### PR DESCRIPTION
**Pull request by oanaianc**
… join

### CODE REVIEW

**Review by raluca8x8 — DISMISSED** [review]

File: doc/example_xmpp_envs.conf [hunk 1]
```diff
@@ -11,6 +11,9 @@
             // The base XMPP domain
             xmpp-domain = "xmpp-domain"
 
+            // An (optional) base url the Jibri will join if it is set
+            base-url = "http://meet.jit.si"
```
Comment (saghul) [inline doc/example_xmpp_envs.conf:15]: Can you please leave it commented out so people don't enable it by mistake?

Also use https://meet.example.com to make it obvious it needs changing.

**Review by saghul — COMMENTED** [review]
This is awesome, thanks for the quick turnaround! Left a minor comment.

File: doc/example_xmpp_envs.conf (hunk 1 above)
Comment (oanaianc) [reply doc/example_xmpp_envs.conf]: done

**Review by raluca8x8 — APPROVED** [review]

**Review by saghul — APPROVED** [review]

File: src/main/kotlin/org/jitsi/jibri/config/JibriConfig.kt [hunk 2]
```diff
@@ -71,6 +71,11 @@ data class XmppEnvironmentConfig(
      */
     @JsonProperty("xmpp_domain")
     val xmppDomain: String,
+    /**
+     * The baseUrl we'll use, if set, to join the call
+     */
+    @JsonProperty("baseUrl")
+    val baseUrl: String?,
```
Comment (bbaldino) [inline src/main/kotlin/org/jitsi/jibri/config/JibriConfig.kt:78]: We shouldn't add new properties here, as this config is deprecated

**Comment by bbaldino** [comment]
Sorry for the late comment, I was out of town, but it's just a minor one anyway.  Glad this ended up being able to be implemented so easily (I thought it was going to require Jicofo changes).

**Comment by saghul** [comment]
Thanks for checking anyway! This helped get Docker working nicely indeed!

File: src/main/kotlin/org/jitsi/jibri/config/JibriConfig.kt [hunk 3]
```diff
@@ -122,6 +127,9 @@ fun com.typesafe.config.Config.toXmppEnvironment(): XmppEnvironmentConfig =
         name = getString("name"),
         xmppServerHosts = getStringList("xmpp-server-hosts"),
         xmppDomain = getString("xmpp-domain"),
+        baseUrl = if (hasPath("base-url")) {
```
Comment (Tinkesh-Kumar) [inline src/main/kotlin/org/jitsi/jibri/config/JibriConfig.kt:130]: Can we allow this to set at run time ?, Similar to HttpAPI.

**Pull request merged**